### PR TITLE
DCWL-1649 Improve error logging to show error codes being received downstream/sent back

### PR DESCRIPTION
**Pull request by kevin-co-hmrc**
Consolidate error logging from downstream so that we can see:

a) what error codes we are receiving
b) what error codes we are sending back to the client

### CODE REVIEW

File: app/uk/gov/hmrc/customs/declarations/information/services/declarationsInformationServices.scala [hunk 1]
```diff
@@ -193,59 +187,38 @@ abstract class DeclarationService @Inject()(override val apiSubFieldsConnector:
...
+                    s"[${errorResponse.httpStatusCode}] and response body: [${errorResponse.XmlResult.body.asInstanceOf[HttpEntity.Strict].data.utf8String}]")
+                  Left(errorResponse.XmlResult.withConversationId)
+                case FORBIDDEN =>
+                  logger.warn(s"declaration [$endpointName] call failed with backend http status code of [403] so returning to consumer [403]")
+                  Left(ErrorPayloadForbidden.XmlResult.withConversationId)
+                case unexpectedStatus =>
+                  logger.error(s"declaration [$endpointName] call failed with unexpected backend http status code of [$unexpectedStatus] so returning to consumer [500]")
+                  Left(ErrorInternalServerError.XmlResult.withConversationId)
+              }
+          }
```
Comment (IanWood-HMRC) [inline app/uk/gov/hmrc/customs/declarations/information/services/declarationsInformationServices.scala:221]: Is it possible to introduce a method to handleResult to reduce duplication as in the example [here] (https://github.com/hmrc/customs-inventory-linking-imports/blob/a77d15aa952200aaa0c5d9cbc59348dc4beaf430/app/uk/gov/hmrc/customs/inventorylinking/imports/services/MessageSender.scala#L91)

Comment (kevin-co-hmrc) [reply app/uk/gov/hmrc/customs/declarations/information/services/declarationsInformationServices.scala:218]: Done, though it's not how I prefer to do it as it introduces an extra level of indirection just to save a couple of lines.

**Review by ed1286 — APPROVED** [review]

File: app/uk/gov/hmrc/customs/declarations/information/services/declarationsInformationServices.scala [hunk 2]
```diff
@@ -54,7 +50,7 @@ class DeclarationStatusService @Inject()(statusResponseFilterService: StatusResp
       case "cds60001" => backendCDS60001NotFoundResponse
       case "cds60002" => backendCDS60002SearchInvalidResponse
       case "cds60003" => backendCDS60003InternalServerErrorResponse
-      case _          => ErrorInternalServerError
+      case _ => ErrorInternalServerError
```
Comment (IanWood-HMRC) [inline app/uk/gov/hmrc/customs/declarations/information/services/declarationsInformationServices.scala:53]: can these have the same indention as before?

File: app/uk/gov/hmrc/customs/declarations/information/services/declarationsInformationServices.scala [hunk 3]
```diff
@@ -84,7 +80,7 @@ class DeclarationVersionService @Inject()(versionResponseFilterService: VersionR
       case "cds60002" => backendCDS60002MrnInvalidResponse
       case "cds60003" => backendCDS60003InternalServerErrorResponse
       case "cds60011" => backendCDS60011SubmissionChannelInvalidResponse
-      case _          => ErrorInternalServerError
+      case _ => ErrorInternalServerError
```
Comment (IanWood-HMRC) [inline app/uk/gov/hmrc/customs/declarations/information/services/declarationsInformationServices.scala:83]: can these have the same indention as before?

File: app/uk/gov/hmrc/customs/declarations/information/services/declarationsInformationServices.scala [hunk 4]
```diff
@@ -127,7 +123,7 @@ class DeclarationSearchService @Inject()(searchResponseFilterService: SearchResp
       case "cds60010" => backendCDS60010GoodsLocationCodeInvalidResponse
       case "cds60011" => backendCDS60011SubmissionChannelInvalidResponse
       case "cds60012" => backendCDS60012PageNumberInvalidResponse
-      case _          => ErrorInternalServerError
+      case _ => ErrorInternalServerError
```
Comment (IanWood-HMRC) [inline app/uk/gov/hmrc/customs/declarations/information/services/declarationsInformationServices.scala:126]: can these have the same indention as before?

File: app/uk/gov/hmrc/customs/declarations/information/services/declarationsInformationServices.scala [hunk 5]
```diff
@@ -156,7 +152,7 @@ class DeclarationFullService @Inject()(fullResponseFilterService: FullResponseFi
       case "cds60002" => backendCDS60002MrnInvalidResponse
       case "cds60003" => backendCDS60003InternalServerErrorResponse
       case "cds60011" => backendCDS60011SubmissionChannelInvalidResponse
-      case _          => ErrorInternalServerError
+      case _ => ErrorInternalServerError
```
Comment (IanWood-HMRC) [inline app/uk/gov/hmrc/customs/declarations/information/services/declarationsInformationServices.scala:155]: can these have the same indention as before?

File: app/uk/gov/hmrc/customs/declarations/information/services/declarationsInformationServices.scala (hunk 2 above)
Comment (kevin-co-hmrc) [reply app/uk/gov/hmrc/customs/declarations/information/services/declarationsInformationServices.scala:53]: Not sure how this happened, but will do.

File: app/uk/gov/hmrc/customs/declarations/information/services/declarationsInformationServices.scala (hunk 3 above)
Comment (kevin-co-hmrc) [reply app/uk/gov/hmrc/customs/declarations/information/services/declarationsInformationServices.scala:83]: Not sure how this happened, but will do.

File: app/uk/gov/hmrc/customs/declarations/information/services/declarationsInformationServices.scala (hunk 4 above)
Comment (kevin-co-hmrc) [reply app/uk/gov/hmrc/customs/declarations/information/services/declarationsInformationServices.scala:126]: Not sure how this happened, but will do.

File: app/uk/gov/hmrc/customs/declarations/information/services/declarationsInformationServices.scala (hunk 5 above)
Comment (kevin-co-hmrc) [reply app/uk/gov/hmrc/customs/declarations/information/services/declarationsInformationServices.scala:155]: Not sure how this happened, but will do.

**Pull request merged**